doschema/merger.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of DoSchema.
+# Copyright (C) 2016 CERN.
+#
+# DoSchema is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# DoSchema is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with DoSchema; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Merger module."""
+
+
+class JsonSchemaCompatibilityChecker(object):
+    """Class for checking compatibility between schemas."""
+
+    def __init__(self, schema_id, curr_field, curr_schema, ignore_index=True):
+        """Constructor.
+
+        :param schema_id: Index of schema that is currently proceeded
+        :param curr_field: Tuple with path to currently proceeded field
+        :param curr_schema: Schema or subschema that is currently proceeded
+        """
+        self.schema_id = schema_id
+        self.curr_field = curr_field
+        self.curr_schema = curr_schema
+        self.ignore_index = ignore_index
+
+    def traverse(self, fields_types_dict):
+        """Go through the schema and retrieves his particular fields.
```

Comment:
Comments should be imperative: `s/retrieve/retrieves/`. Also who is designated by `his`?